frontend/app/components/base-form-component.js

```diff
@@ -0,0 +1,45 @@
+import { action } from "@ember/object";
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { inject as service } from "@ember/service";
+
+export default class BaseFormComponent extends Component {
+  @service intl;
+
+  @tracked
+  record;
+
+  async beforeSubmit() {}
```

Comment:
would it make sense to leave some comments in this file?
maybe with an example when one might use one of those hooks?